libhttpsimple-bindings/src/main/scala/libhttpsimple/LibHttpSimple.scala

```diff
@@ -82,9 +73,19 @@ object LibHttpSimple {
         if (errorCode == 0) {
           val httpStatus = nativebinding.httpsimple.get_http_status(http_response_struct).cast[Long]
           val rawHttpResponse = native.fromCString(nativebinding.httpsimple.get_raw_http_response(http_response_struct))
-          val (header, body) = rawHttpResponseToHttpHeadersAndBody(rawHttpResponse)
-
-          Success(HttpResponse(httpStatus, header, body))
+          val (headers, body) = rawHttpResponseToHttpHeadersAndBody(rawHttpResponse)
+          val hs = HttpHeaders(headers)
+
+          if (followRedirects && hs.contains("Location")) {
+            val location = hs("Location")
+
+            if (visitedUrls.contains(location))
```

Comment:
Can I also suggest checking `visitedUrls <= MaxRedirect`, and let's say we have `MaxRedirect` declared as constant in the `LibHttpSimple`?